src/v/kafka/server/tests/quota_manager_test.cc

```diff
@@ -15,33 +15,43 @@
 #include <seastar/core/sleep.hh>
 #include <seastar/core/sstring.hh>
 #include <seastar/testing/thread_test_case.hh>
+#include <seastar/util/defer.hh>
 
 #include <boost/test/tools/old/interface.hpp>
 #include <boost/test/unit_test.hpp>
 
-#include <random>
-
 using namespace std::chrono_literals;
 
 const static auto client_id = "franz-go";
 
-template<typename F>
-ss::future<> run_quota_manager_test(F test_core) {
+struct fixture {
     kafka::quota_manager::client_quotas_t buckets_map;
     ss::sharded<kafka::quota_manager> sqm;
-    co_await sqm.start(std::ref(buckets_map));
-    co_await sqm.invoke_on_all(&kafka::quota_manager::start);
+
+    ss::future<> start() {
+        co_await sqm.start(std::ref(buckets_map));
+        co_await sqm.invoke_on_all(&kafka::quota_manager::start);
+    }
+
+    ss::future<> stop() { co_await sqm.stop(); }
+};
+
+template<typename F>
+ss::future<> run_quota_manager_test(F test_core) {
+    fixture f;
+    co_await f.start();
 
     // Run the core of the test now that everything's set up
-    co_await ss::futurize_invoke(test_core, sqm);
+    co_await ss::futurize_invoke(test_core, f.sqm);
 
-    co_await sqm.stop();
+    co_await f.stop();
 }
 
 template<typename F>
 ss::future<> set_config(F update) {
     co_await ss::smp::invoke_on_all(
       [update{std::move(update)}]() { update(config::shard_local_cfg()); });
+    co_await ss::sleep(std::chrono::milliseconds(1));
```

Comment:
I seem to recall something about seastar randomizing task queues in debug mode...would that have an effect here?

Reply:
> I seem to recall something about seastar randomizing task queues in debug mode...would that have an effect here?

Yeah, I chose `sleep` instead of `yield` with roughly that reasoning.